platform/msm_shared/scm.c

```diff
@@ -1077,6 +1077,12 @@ void scm_elexec_call(paddr_t kernel_entry, paddr_t dtb_offset)
 		scm_arg.x2 = (uint32_t ) &param;
 		scm_arg.x3 = sizeof(el1_system_param);
 
+		/* First try a hypervisor call to jump to EL2 if supported */
+		scm_arg.hvc = true;
+		scm_call2(&scm_arg, NULL);
+
+		dprintf(INFO, "Falling back to SCM call\n");
+		scm_arg.hvc = false;
```

Comment:
This will crash on platforms that don't have hyp partition (or otherwise don't allow unknown hvc calls) so we didn't include el2 support until now, pending changes in qhypstub. Unfortunately neither me nor @stephan-gh had time to fix this yet...